### ISSUE TITLE
consistent-function-scoping false positive modifying property of variable

### ISSUE THREAD

**Issue by kevinoid** [sindresorhus/eslint-plugin-unicorn#374]
Running ESLint 6.4.0 with eslint-plugin-unicorn 11.0.0 and `"extends": "plugin:unicorn/recommended"` on a file with the following reduced example code:
```js
'use strict';

module.exports = function recordErrors(eventEmitter, stateArgument) {
  const stateVariable = stateArgument;
  function onError(error) {
    stateVariable.inputError = error;
  }

  eventEmitter.once('error', onError);
};
```

gives the following output:

    /tmp/consistent-function-scoping-issue/issue.js
      5:3  error  Move function to the outer scope  unicorn/consistent-function-scoping
    
    ✖ 1 problem (1 error, 0 warnings)

I believe this is incorrect since `onError` can not be moved to the global scope, since it accesses `stateVariable`, which is only defined in the function scope.

Thanks,
Kevin

**Comment by kevinoid** [comment]
I just noticed that a similar issue occurs when calling functions which must be defined in the same scope:

```js
'use strict';

module.exports = function recordErrors(eventEmitter, stateArgument) {
  function onError(error) {
    stateArgument.inputError = error;
  }
  function onError2(error) {
    onError(error);
  }

  eventEmitter.once('error', onError2);
};
```

Since `onError2` can't be moved to the global scope because it calls `onError` which can't be moved to the global scope.

If you'd like me to open a separate issue to track this variant, let me know.

Thanks again,
Kevin

**Comment by MrHen** [comment]
> If you'd like me to open a separate issue to track this variant, let me know.

Yes, please do. I didn't think about that scenario at all.

**Comment by kevinoid** [comment]
> > If you'd like me to open a separate issue to track this variant, let me know.
> 
> Yes, please do. I didn't think about that scenario at all.

Sure thing.  Moved to #375.
